README.md

```diff
@@ -129,6 +130,51 @@ Properties can be provided by (order makes precedence):
   
 4. Mixed approach.
 
+### Validating properties
+
+Configuring of project properties can be enhanced by providing properties definitions which can be used for property value validation, e.g.:
+```kotlin
+fork {
+    properties(mapOf(
+            "enableSmbClient" to {
+                type = CHECKBOX
```

Comment:
I prefer `PropertyType.CHECKBOX` with import or introduce a method `checkbox()` in a same manner as `optional()` that mutes internal state